### PR TITLE
Ticket/9757

### PR DESCRIPTION
**Pull request by mariovalney**
Fixes:

* Fix layout for big "Install now" translations (like the Brazilian "Instalar agora").
* Fix layout for mobile help text.
* Accept only zip as plugin upload is not able to handle only-file plugins.

Trac ticket: https://core.trac.wordpress.org/ticket/9757

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-admin/includes/class-plugin-installer-skin.php [hunk 1]
```diff
@@ -138,4 +154,67 @@ public function after() {
...
+		$new_plugin_data     = array_filter( $this->upgrader->new_plugin_data );
+
+		$table  = '<table class="compare-plugins-table"><tbody>';
+		$table .= '<tr><th><th/><th>' . esc_html( __( 'Current' ) ) . '<th/><th>' . esc_html( __( 'Uploaded' ) ) . '</th></tr>';
+
+		foreach ( $rows as $field => $label ) {
+			$old_value = ! empty( $current_plugin_data[ $field ] ) ? $current_plugin_data[ $field ] : '-';
+			$new_value = ! empty( $new_plugin_data[ $field ] ) ? $new_plugin_data[ $field ] : '-';
+
+			$table .= '<tr><td>' . $label . '<td/><td>' . esc_html( $old_value ) . '<td/><td>' . esc_html( $new_value ) . '</td></tr>';
```
Comment (azaozz) [inline src/wp-admin/includes/class-plugin-installer-skin.php:204]: HTML closing tags typos? :)

File: src/wp-admin/update.php [hunk 2]
```diff
@@ -163,7 +163,12 @@
 		$type  = 'upload'; // Install plugin type, From Web or an Upload.
 
 		$upgrader = new Plugin_Upgrader( new Plugin_Installer_Skin( compact( 'type', 'title', 'nonce', 'url' ) ) );
-		$result   = $upgrader->install( $file_upload->package );
+
+		$args = array(
+			'overwrite_package' => ! empty( $_GET['overwrite'] ),
```
Comment (azaozz) [inline src/wp-admin/update.php:168]: May be good to make the query arg a bit longer/more specific? Perhaps something like `$_GET['overwrite'] === 'ovewrite-uploaded-plugin'`.

File: src/wp-admin/includes/class-plugin-installer-skin.php (hunk 1 above)
Comment (mariovalney) [reply src/wp-admin/includes/class-plugin-installer-skin.php]: :open_mouth:

**Comment by mariovalney** [comment]
TODO:

- [ ] Replace "Compare and retry" with the "same plugin/theme" message.

**Pull request closed without merging**